builders/invoke-workflow.psm1

```diff
@@ -0,0 +1,52 @@
+function Invoke-Workflow {
+    param (
+        [string] $Version,
+        [bool] $PublishRelease
+    )
+    
+    $payload = @{
+        "ref" = "main"
+        "inputs" = @{
+            "VERSION" = "$Version"
+            "PUBLISH_RELEASES" = "$PublishRelease"
+        }
+    } | ConvertTo-Json
+    $headers = @{
+        Authorization="Bearer $env:PERSONAL_TOKEN"
+    }
+
+    Invoke-RestMethod -uri "$env:GITHUB_API_URL/repos/$env:GITHUB_REPOSITORY/actions/workflows/python-builder.yml/dispatches" -method POST -headers $headers -body $payload
+
+    $result = [PSCustomObject]@{
+        Version = $version
+        Conclusion = "failure"
+        Url = "Not run"
+    }
+        
+    while ($workflowToCheck.status -ne "completed") {
+        while (!$workflowToCheck) {
```

Comment:
Can we split these two `while` loops as it looks more simple? Currently, `$workflowToCheck.status` will not exist when we check it for the first time
```pwsh
 while (-not $workflowToCheck) {
# wait for workflow to be triggered
}

while ($workflowToCheck.status -ne "completed") {
# check the workflow status
}
```

Reply:
Done